src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -866,6 +868,25 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
     }
 
+    /**
+     * In the case we follow an elected master the new cluster state needs to have the same elected master and
+     * the new cluster state version needs to be equal or higher than our cluster state version. If either conditions
+     * are true then the cluster state is dated and we should ignore it.
+     */
+    private boolean newClusterStateDated(ClusterState currentState, ClusterState newClusterState) {
```

Comment:
can we rename this to shouldIgnoreNewClusterState? it's not only about being dated.